airbyte-integrations/connectors/source-iterable/unit_tests/test_api.py

```diff
@@ -0,0 +1,172 @@
+        )
+        resp = requests.get("https://api.iterable.com/api/1/foobar")
+
+        records = stream.parse_response(response=resp)
+
+        assert list(records) == [{"id": 1, "createdAt": pendulum.parse("2022", strict=False)}]
+
+
+def test_list_users_parse_response():
+    stream = ListUsers(authenticator=NoAuth())
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            responses.GET,
+            "https://api.iterable.com/lists/getUsers?listId=100",
+            body="user100",
+            status=200,
+            content_type="application/json",
+        )
+        resp = requests.get("https://api.iterable.com/lists/getUsers?listId=100")
+
+        records = stream.parse_response(response=resp)
+
+        assert list(records) == [{"email": "user100", "listId": 100}]
+
+
+def test_campaigns_metrics_parse_response():
+
+    stream = CampaignsMetrics(authenticator=NoAuth(), start_date="2019-10-10T00:00:00")
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            responses.GET,
+            "https://api.iterable.com/lists/getUsers?listId=100",
+            body="""a,b,c,d
+1, 2,, 3
+6,, 1, 2
+""",
+            status=200,
+            content_type="application/json",
+        )
+        resp = requests.get("https://api.iterable.com/lists/getUsers?listId=100")
+
+        records = stream.parse_response(response=resp)
+
+        assert list(records) == [
+            {"data": {"a": 1, "b": 2, "d": 3}},
+            {"data": {"a": 6, "c": 1, "d": 2}},
+        ]
+
+
+def test_iterable_stream_parse_response():
+    stream = Lists(authenticator=NoAuth())
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            responses.GET,
+            "https://api.iterable.com/lists/getUsers?listId=100",
+            json={"lists": [{"id": 1}, {"id": 2}]},
+            status=200,
+            content_type="application/json",
+        )
+        resp = requests.get("https://api.iterable.com/lists/getUsers?listId=100")
+
+        records = stream.parse_response(response=resp)
+
+        assert list(records) == [{"id": 1}, {"id": 2}]
+
+
+def test_iterable_stream_backoff_time():
+    stream = Lists(authenticator=NoAuth())
+    assert stream.backoff_time(response=None) == stream.BACKOFF_TIME_CONSTANT
+
+
+def test_iterable_export_stream_backoff_time():
+    stream = Users(authenticator=NoAuth(), start_date="2019-10-10T00:00:00")
+    assert stream.backoff_time(response=None) is None
+
+
+@pytest.mark.parametrize(
+    "current_state,record_date,expected_state",
+    [
+        ({}, "2022", {"profileUpdatedAt": "2022-01-01T00:00:00+00:00"}),
+        ({"profileUpdatedAt": "2020-01-01T00:00:00+00:00"}, "2022", {"profileUpdatedAt": "2022-01-01T00:00:00+00:00"}),
+        ({"profileUpdatedAt": "2022-01-01T00:00:00+00:00"}, "2020", {"profileUpdatedAt": "2022-01-01T00:00:00+00:00"}),
+    ],
+)
+def test_get_updated_state(current_state, record_date, expected_state):
```

Comment:
I think this tests is redundant because `Users` is not overriding `get_updated_state`. Please make sure its parent `get_updated_state`'s is tested.